genshin/client/manager.py

```diff
@@ -76,6 +78,17 @@ def user_id(self) -> typing.Optional[int]:
         """
         return None
 
+    @property
+    def proxy(self) -> typing.Optional[yarl.URL]:
+        """Get proxy setting"""
+        return self._proxy
+
+    @proxy.setter
+    def proxy(self, proxy: yarl.URL) -> None:
+        """Set proxy, aiohttp only support [http, https, ws, wss] proxy."""
+        if str(proxy.scheme) in {"https", "http", "ws", "wss"}:
+            self._proxy = proxy
```

Comment:
Either raise an error telling the user they passed in an invalid proxy or keep it as a plain attribute.